src/formatDate.js

```diff
@@ -8,7 +8,51 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(/[-/.]/);
+  const valueMap = fromFormat.reduce((accumulator, format, index) => {
+    accumulator[format] = dateParts[index];
+
+    return accumulator;
+  }, {});
+
+  const yy = Number(valueMap['YY']);
+  const year =
+    valueMap['YYYY'] ||
+    (yy >= 0 && yy <= 99
+      ? yy < 30
+        ? `20${String(yy).padStart(2, '0')}`
+        : `19${String(yy).padStart(2, '0')}`
+      : valueMap['YY']);
+
+  const month = valueMap['MM'] ? Number(valueMap['MM'], 10) - 1 : 0;
+  const day = valueMap['DD'] ? Number(valueMap['DD'], 10) : 1;
+
+  if (!year) {
+    return null;
+  }
+
+  const parsedDate = new Date(year, month, day);
+
+  if (isNaN(parsedDate.getTime())) {
```

Comment:
Creating a `Date` object and checking for invalid dates is beyond the scope of the task. The task is to reformat the date string, not to validate it. You should remove the creation of the `Date` object and the associated validity check.